portfolio/src/main/java/com/google/sps/servlets/CommentDeletionServlet.java

```diff
@@ -0,0 +1,29 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that handle comments deletion. */
+@WebServlet("/delete-comment")
```

Comment:
Should this be called something like `delete-all-comments` instead? `delete-comment` feels like it should take in a single comment ID to delete.

Reply:
I think instead it should just be a DELETE request to "/comments"

Reply:
I am not sure about how much responsibility a servlet should take. Should a single servlet handle all CRUD operations for comment or should I create servlets for each operations? The instructions ask us to create a servlet for /delete-comment here, so I just put this /delete-comment. I was thinking about using this servlet to handle deletion of a single comment as well, but actually I am not sure what is the proper way to do this.

Reply:
I didn't know about the instructions in the walkthrough.
In general it's better to make design choices with API in mind rather than implementation. It makes sense here to write it in REST style since it's clearly for manipulating just one type of resource - comments. Most of Google's APIs are written this way. There's also a pretty good style guide for writing APIs at https://cloud.google.com/apis/design. (It's written with another framework different from REST (gRPC) in mind but the ideas are the same).
I don't think it's an issue if one servlet serves all the CRUD operations. As far as implementation is concerned, if it becomes too complex, you can always write a separate class for handling deletion and call that class from the servlet.

Reply:
Thank you. I see. For now, I just let this servlet handle deleting all comments only. Later may add other functions to the /comments servlet.